WordPress/gutenberg
# Components: refactor `RangeControl` to pass `exhaustive-deps`

## What?
Updates the `RangeControl` component to satisfy the `exhaustive-deps` eslint rule

## Why?
Part of the effort in https://github.com/WordPress/gutenberg/pull/41166 to apply `exhuastive-deps` to the Components package

## How?
`tooltip.tsx`:
- `inputRef` should always be a ref object passed in by the consumer, even React can't tell that it is. As a ref, it'll stay stable and a safe dependency.

`utils.ts`:
-  **useControlledRangeValue** `setInternalState` is declared by running the `useControlledState` hook. I didn't notice any adverse effects to this new dependency, and the underlying function is also being memoized in #44161
- **useDebouncedHoverInteraction**
    - it seems that this hook was intended to show the tooltip on mouse enter (move) and hide the tooltip on mouse leave, but currently the `RangeControl` component is showing/hiding the `Tooltip` component on focus/blur.
    - Therefore, it appears that the `useDebouncedHoverInteraction` hook was simply out of date and can be deleted without causing any consequences
    - The `onShowTooltip` and `onHideTooltip` props from `InputRangeProps` were also deleted since they were only used when passed to the `useDebouncedHoverInteraction` prop, and are therefore not used anymore

## Testing Instructions
1. From your local Gutenberg directory, run `npx eslint --rule 'react-hooks/exhaustive-deps: warn' packages/components/src/range-control`
2. Confirm that the linter returns no errors
3. Confirm unit tests still pass
4. Run Storybook locally, confirm the components stories and/or docs still work as expected
